internal/jar/manifest.go

```diff
@@ -0,0 +1,43 @@
+//
+// Copyright 2022.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package jar
+
+import "fmt"
+
+const (
+	implementationTitle   = "Implementation-Title"
+	implementationVersion = "Implementation-Version"
+)
+
+type Manifest struct {
+	properties map[string]string
+}
+
+func (m *Manifest) GetImplementationVersion() (string, error) {
+	if val, ok := m.properties[implementationVersion]; !ok {
+		return "", fmt.Errorf("Implementation Version not found in manifest")
```

Comment:
```suggestion
		return "", errors.New("Implementation-Version not found in manifest")
```

Reply:
Maybe pull this out into a separate function that returns an error for missing keys, but it's not a big deal.